assets/js/modules/analytics-4/components/common/PropertySelect.js

```diff
@@ -88,6 +88,37 @@ export default function PropertySelect( {
 	);
 
 	const { selectProperty } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const measurementIDs = {};
+
+	useSelect( ( select ) => {
+		if ( ! properties.length ) {
+			return null;
+		}
+
+		const propertyMeasurementMapping = properties.map( ( property ) => {
+			const currentPropertyID = property._id;
+
+			const dataStream =
+				select( MODULES_ANALYTICS_4 ).getWebDataStreamsBatch(
+					currentPropertyID
+				);
+
+			if ( ! Object.keys( dataStream ).length ) {
+				isLoading = true;
+				return null;
+			}
+
+			const measurementID =
+				dataStream[ currentPropertyID ][ 0 ].webStreamData
+					.measurementId; // eslint-disable-line sitekit/acronym-case
+			isLoading = false;
+			return ( measurementIDs[ currentPropertyID ] = measurementID );
+		} );
+
+		return propertyMeasurementMapping;
+	} );
+
```

Comment:
The better API would be to return the mapped object from the `useSelect` hook and use it like this:
```suggestion
	const measurementIDs = useSelect( ( select ) => {
		if ( ! properties.length ) {
			return null;
		}

		return properties.reduce( ( acc, property ) => {
			const currentPropertyID = property._id;

			const dataStream =
				select( MODULES_ANALYTICS_4 ).getWebDataStreamsBatch(
					currentPropertyID
				);

			if ( ! Object.keys( dataStream ).length ) {
				isLoading = true;
				return null;
			}

			const measurementID =
				dataStream[ currentPropertyID ][ 0 ].webStreamData
					.measurementId; // eslint-disable-line sitekit/acronym-case
			isLoading = false;
			return {
				...acc,
				[ currentPropertyID ]: measurementID,
			};
		}, {} );
	} );
```

Reply:
Yes, let's not run `useSelect` calls for side-effects as in the original code. @hussain-t's suggestion/approach makes sense: we should always try to use values returned from hooks and not have a `useSelect` mutate outside variables; it's harder to keep track of.

Reply:
Just noticing one other thing: I don't know if the `isLoading` mutation in there is exactly what we want—maybe that should be a state variable?

Reply:
@tofumatt, my point is not to set the loading state in that place. Please refer to this [comment](https://github.com/google/site-kit-wp/pull/5849#discussion_r994457037).